Crud

hugoMontevideo opened the pull request:
portfoliobyId() dans l'api

hugomontes24 committed:
Feat(Authentication): debugging authentication.
Adding properties to User(UserDetails)
branch feat/authentication #16

hugomontes24 committed:
Feat(crud) : Portfolio methods getAll and getById.
branch feat/crud #18

hugomontes24 committed:
working in process

hugomontes24 committed:
Feat(crud): Getting portfolio by Id.
branch bac #18

hugomontes24 committed:
Feat(login): saving user on sessionStorage.
Description: Login (email) and token on sessionStorage.
branch bac #14

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/login/login-user.interface.ts:
```diff
@@ -1,5 +1,5 @@
-    email: string,
+    login: string,
     token: string,
     conButton: string
```
connexionStatus en boolean

hugomontes24 replied:
ok

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list/portfolio-list.component.ts:
```diff
@@ -1,42 +1,36 @@
-              };
   loginUser: LoginUser ={
-                email:"",
+                login:"",
```
je preferais email ou username

hugomontes24 replied:
ok

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/services/login.service.ts:
```diff
@@ -16,43 +17,42 @@ export class LoginService {
+      conButton: "Connexion"
     } ;
-    sessionStorage.removeItem('currentUser');
+    sessionStorage.clear();
```
evite le clear par ce que ça vide tout remove les un par un

hugomontes24 replied:
ok

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/services/session-storage.service.ts:
```diff
@@ -18,14 +16,13 @@ export class SessionStorageService {
     sessionStorage.setItem('login', login);
   }
-  public getLogin = () => {
+  public getLogin = (): string | any => {
```
ce serait mieux que getLogin ne renvoi que string

hugomontes24 replied:
ok

hugomontes24 committed:
Feat(crud_skill): addSkill method
Description: adding a skill from front
branch bac #23

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/controllers/skills/SkillController.java:
```diff
@@ -1,22 +1,32 @@
     @Autowired
     SkillService skillService;
+    @Autowired
+    PortfolioService portfolioService;
```
Mieux vaut avoir un service qui gère les deux

Mekouar-Ayoub replied:
parce que la logique doit se situer dans service pas autre part

hugomontes24 replied:
ok

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioServiceRequestModel.java:
```diff
@@ -1,9 +1,15 @@
 package com.simplon.easyportfolio.api.services.portfolios;
```
Je comprend pas que tu ai deux classes services pour portfolio

hugomontes24 replied:
j'avais une classe avec un id optionnal, mais je me suis rendu compte plus tard que j'avais besoin d'une classe avec un Long id et une autre sans id (creation de portfolio)

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/skills/SkillService.java:
```diff
@@ -12,6 +16,11 @@
     SkillRepository skillRepository;
+
+    @Autowired
+    PortfolioRepository portfolioRepository;
```
Voilà ton skill service gere bien deux repository donc c'est le seul qui doit apparaitre dans ton controller

hugomontes24 replied:
ok

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/education/education.interface.ts:
```diff
@@ -0,0 +1,9 @@
+export interface Education {
+    id: number,
```
C'est un dto ou une interface de composant ?

hugomontes24 replied:
une interface

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.scss:
```diff
@@ -0,0 +1,15 @@
+h5{
```
evite les selecteurs par type dans les composants, seulement dans app.component.css

hugomontes24 replied:
ok

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.ts:
```diff
@@ -11,8 +15,7 @@ import { PortfolioService } from '../../services/portfolio.service';
   table: string = 'portfolios';
-  id: number|any;
-  currentUser!: User;
+  portfolioId: string|any;
```
mmmm any

hugomontes24 replied:
ok

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.ts:
```diff
@@ -21,40 +24,79 @@ export class PortfolioListItemComponent implements OnInit {
+    } 
+    legend!: string;
+    skillEdit: Skill = {
+      id:null,
```
initialise le plutot à -1

hugomontes24 replied:
ok

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.ts:
```diff
@@ -21,40 +24,79 @@ export class PortfolioListItemComponent implements OnInit {
-      if(this.id != null){ // *************** todo  //
-        this.getPortfolioById(this.table, 1);
+      this.portfolioId =  this.route.snapshot.paramMap.get('id');
+      this.loginUser.login = this.storageService.getLogin();
```
j'aurais prefere que ton service renvoi les deux données d'un coup

hugomontes24 replied:
ok

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.ts:
```diff
@@ -21,40 +24,79 @@ export class PortfolioListItemComponent implements OnInit {
+      this.portfolioId =  this.route.snapshot.paramMap.get('id');
+      this.loginUser.login = this.storageService.getLogin();
+      this.loginUser.conButton = this.storageService.getConButton();  
+      if( this.loginUser.login != ""){
```
cette condition devrait peut etre etre gérée dans le service

hugomontes24 replied:
ok

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/skill/skill.component.scss:
```diff
@@ -0,0 +1,21 @@
+p{
```
evite les selecteurs par type dans les composants

hugomontes24 replied:
ok

hugomontes24 committed:
Feat(Crud Skill): Deleting a skill from portfolio.
branch bac #23

hugomontes24 committed:
Feat(Crud Experience): addExperience method.
branch bac #24

hugomontes24 committed:
Feat(Crud Education): addEducation method.
Description: Passing a date to the backend -> Date object on the frontend, LocalDate object on the backend
branch bac #26

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/controllers/auth/SecurityController.java:
```diff
@@ -21,24 +21,24 @@ public class SecurityController {
     public ResponseEntity<AuthResponseDto> register(@RequestBody AuthRequestDto dto) throws AccountExistsException {
-        UserDetails user = userService.save(dto.getUsername(),
+        UserDetails user = userService.save(dto.getEmail(),
                 dto.getPassword());
```
Je t'ai mis des commentaires directement sur le code pas dans la pull request

hugoMontevideo replied:
ok.vu

hugomontes24 committed:
Feat(Crud Project): AddProject, deleteProject methods.
branch bac #

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/controllers/portfolios/PortfolioController.java:
```diff
@@ -70,7 +70,7 @@ public ResponseEntity<String> updateFolio(@PathVariable("id") Long id,
     public ResponseEntity<String> delete(@PathVariable Long id){
         if(portfolioService.findById(id) != null){
```
try catch plutot

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/controllers/projects/ProjectController.java:
```diff
@@ -1,18 +1,65 @@
+    public boolean add(@RequestBody ProjectDTO projectDTO){
+        ProjectServiceRequestModel projectServiceRequestModel =
+                mapper.projectDtoToServiceRequestModel(projectDTO);
+        PortfolioServiceModel portfolio = new PortfolioServiceModel(projectDTO.getPortfolioId());
```
je comprend pas pourquoi tu fais ça, il mappe pas automatiquement mapstruct ?

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioService.java:
```diff
@@ -83,7 +89,43 @@ public void delete(Long id) {
+        // Table : PROJECT   *****************
+    // add Project
+    public boolean addProject(ProjectServiceRequestModel projectServiceRequestModel) {
+        ProjectRepositoryModel project = mapper.projectServiceRequestToRepositoryModel(projectServiceRequestModel);
```
tu arrive à mapper les listes à l'intérieur ?

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioService.java:
```diff
@@ -83,7 +89,43 @@ public void delete(Long id) {
+        // TODO *** finish the method, not working yet ***
+        Optional<ProjectRepositoryModel> projectRepositoryModel = projectRepository.findById(id);
+
+        if (projectRepositoryModel.isEmpty()){
```
j'aime pas trop le isEmpty pourquoi tu teste pas null ?

hugoMontevideo replied:
méthode à finaliser

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioService.java:
```diff
@@ -83,7 +89,43 @@ public void delete(Long id) {
+        }else{
+            //SkillServiceResponseModel item = mapper.skillRepositoryToResponseSvc(skillRepositoryModel.get());
+            //return item;
+            return new ProjectServiceResponseModel();
```
pourquoi tu retourne un objet vide ?

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioService.java:
```diff
@@ -83,7 +89,43 @@ public void delete(Long id) {
+    }
+
+    // delete Project
+    public boolean deleteProject(Long id) {
```
try catch plutot que find puis delete

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioService.java:
```diff
@@ -106,18 +148,44 @@ public boolean addEducation(EducationServiceRequestModel educationRequest) {
+            //SkillServiceResponseModel item = mapper.skillRepositoryToResponseSvc(skillRepositoryModel.get());
+
+            //return item;
+            return new EducationServiceResponseModel();
```
meme chose

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioService.java:
```diff
@@ -106,18 +148,44 @@ public boolean addEducation(EducationServiceRequestModel educationRequest) {
+    // delete Education
+    public boolean deleteEducation(Long id) {
+
+        if(educationRepository.findById(id).isPresent()){
```
try catch

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioService.java:
```diff
@@ -132,7 +200,7 @@ public ExperienceServiceResponseModel findExperienceById(Long id) {
     }
-
+    // delete Education
     public void deleteExperience(Long id) {
```
try catch

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/projects/ProjectServiceRequestModel.java:
```diff
@@ -0,0 +1,19 @@
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class ProjectServiceRequestModel {
```
tu as deux modeles de service ?

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/core/footer/footer.component.html:
```diff
@@ -0,0 +1 @@
+<p>footer works!</p>
```
magnifique

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/education/education.component.ts:
```diff
@@ -8,5 +9,65 @@ import { Education } from './education.interface';
+  legend: string = "Ajouter";
+
+  constructor( private portfolioService: PortfolioService ){}
+  currentEducation: Education = {
```
plutot new Education comme nom

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/education/education.component.ts:
```diff
@@ -8,5 +9,65 @@ import { Education } from './education.interface';
+
+  public onSubmitEducation = ()=>{
+    // // hide the form
+    this.isEducationFormShowing = false; 
```
ou est ce que tu fais la validation du formulaire ?

Mekouar-Ayoub replied:
et l'affichage des erreurs

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/experience/experience.component.ts:
```diff
@@ -8,5 +9,58 @@ import { Experience } from './experience.interface';
 export class ExperienceComponent {
 
-  @Input() experience!:Experience;
+  @Input() experiences!:Experience[];
```
à quoi sert le !

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/experience/experience.component.ts:
```diff
@@ -8,5 +9,58 @@ import { Experience } from './experience.interface';
+
+  legend: string = "Ajouter";
+
+  currentExperience: Experience = {
```
newExperience ou addingExperienceObject

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.html:
```diff
@@ -1,240 +1,50 @@
-                    (click)="onDeleteClick($event)"
+        <div class="actions">
+          <!-- <button class="btn btn-info"
+                    (click)="onEditClick($event)"
```
pourquoi tu lui passe l'event ?

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.html:
```diff
@@ -1,240 +1,50 @@
+                Modifier
+          </button>
+          <button class="btn btn-warning"
+                  (click)="onDeleteClick($event)"
```
ici aussi ?

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.ts:
```diff
@@ -14,11 +10,12 @@ import { Education } from '../education/education.interface';
+
   table: string = 'portfolios';
-  portfolioId!:number;
+  portfolioId:number = -1;
```
tu as porftolio id et un objet porfolio qui a un id, c'est quoi la différence entre les deux

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.ts:
```diff
@@ -14,11 +10,12 @@ import { Education } from '../education/education.interface';
 })
 export class PortfolioListItemComponent implements OnInit {
+
   table: string = 'portfolios';
```
à quoi il te sert ?

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/project.component.ts:
```diff
@@ -9,6 +10,59 @@ import { Project } from './project.interface';
 
-  @Input() project!:Project;
+  @Input() projects!:Project[];
+  @Input() portfolioId!: number;
```
redondance du portfolioId

Mekouar-Ayoub replied:
tu as pas modifié ça visiblement

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/project.component.ts:
```diff
@@ -9,6 +10,59 @@ import { Project } from './project.interface';
+
+  constructor( private portfolioService: PortfolioService ){}
+
+  ngOnChanges(){
```
pourquoi dans le OnChanges ?

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/project.component.ts:
```diff
@@ -9,6 +10,59 @@ import { Project } from './project.interface';
+    // // hide the form
+    this.isProjectFormShowing = false; 
+    this.currentProject.portfolioId = this.portfolioId;  
+    this.portfolioService.addProject('projects' , this.currentProject)
```
vu que mainteant tu as séparé tes composants mieux vaut que tu ai un service pour chacun de tes composants plutot qu'un service qui gère les 3 avec un string qui n'est meme pas une enum et donc prompt aux erreurs

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/services/portfolio.service.ts:
```diff
@@ -41,34 +44,59 @@ export class PortfolioService {
         return this.http.get<any>(`${this.ENV_DEV}/${table}`, { responseType: "json"});
```
get any c'est pas bon

hugomontes24 committed:
Feat(corrections): Corrections of the day
branch crud #30

hugomontes24 committed:
Feat(upload_file): uploading project files
branch bac #32

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioService.java:
```diff
@@ -262,6 +274,34 @@ public void deleteSkill(Long id) {
 
 
+    // UTILS  *****************************************
+    public String uploadPicture(MultipartFile file, String pictureName) throws IOError{
```
met la en private

hugomontes24 committed:
Feat(Add Skill): resolving bugs from previous commits
branch crud  #23

hugomontes24 committed:
Feat(save Skill): renaming addSkill method to saveSkill()
Description: The methods updateSkill and addSkill become saveSkill in PortfolioService.java
branch bac  #23

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/controllers/skills/SkillController.java:
```diff
@@ -38,14 +38,22 @@ public ResponseEntity<SkillGetDTO> findById(@PathVariable Long id){
 
     }
 
+    @PostMapping("/{id}")
```
PutMapping

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/skill/skill.component.ts:
```diff
@@ -23,57 +23,63 @@ export class SkillComponent {
-    error:(err:Error)=>{ console.log("Error while deleting skill.");
-        }
-    }) 
+ public onCloseForm = () => {
```
OnCloseModalForm plutot ?

hugomontes24 committed:
Feat(forms Validation): implementing front 'n back form's validation.
branch bac   #33

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/exceptions/AppHandlerException.java:
```diff
@@ -0,0 +1,29 @@
+package com.simplon.easyportfolio.api.exceptions;
+
```
tu as suivi un tuto ?

hugomontes24 commented:
Salut,

oui


Hugo MACHADO
Developpeur    Web
06 66 26 19 76
________________________________
De : Mekouar Ayoub ***@***.***>
Envoyé : lundi 4 décembre 2023 14:07
À : hugoMontevideo/easyPortfolioProject ***@***.***>
Cc : MACHADO HUGO ***@***.***>; Comment ***@***.***>
Objet : Re: [hugoMontevideo/easyPortfolioProject] Crud (PR #2)


@Mekouar-Ayoub approved this pull request.

________________________________

In api/src/main/java/com/simplon/easyportfolio/api/exceptions/AppHandlerException.java<https://github.com/hugoMontevideo/easyPortfolioProject/pull/2#discussion_r1413845169>:

> @@ -0,0 +1,29 @@
+package com.simplon.easyportfolio.api.exceptions;
+


tu as suivi un tuto ?

—
Reply to this email directly, view it on GitHub<https://github.com/hugoMontevideo/easyPortfolioProject/pull/2#pullrequestreview-1762356920>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASUHOXAT4LJJULWPX53TURLYHXDKJAVCNFSM6AAAAAA7ULE2HSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTONRSGM2TMOJSGA>.
You are receiving this because you commented.Message ID: ***@***.***>

hugomontes24 committed:
Feat(Crud Project): update method.
Description : back 'n Front
branch bac  #29

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/services/portfolios/PortfolioService.java:
```diff
@@ -99,18 +100,61 @@ public ProjectServiceResponseModel saveProject(ProjectServiceRequestModel projec
         projectServiceRequestModel.setPortfolio( Optional.ofNullable(portfolioServiceModel));
+        // verifing date - if date == 1970 01 01 we set it to null
+        if(!isValidDate(projectServiceRequestModel.getDate())){
+            projectServiceRequestModel.setDate(null);
```
evite de set à null utilise les optionnal

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/project.component.ts:
```diff
@@ -53,20 +60,21 @@ export class ProjectComponent {
+                                            data.description,
+                                            data.date,
+                                            data.fileName,
+                                            null,
```
si c'est un objet met le en {} si c'est un type primitif utilise une valeur par défaut

hugomontes24 committed:
Feat(Add Project): Add Project Method. Front 'n Back.
Description: Creating an empty project when adding a project. Scrum Master recommendation
branch bac  #29

Mekouar-Ayoub commented on api/src/main/java/com/simplon/easyportfolio/api/controllers/projects/ProjectController.java:
```diff
@@ -29,23 +32,14 @@ public class ProjectController {
-                mapper.projectDtoToServiceRequestModelAdd(DTO);
+    public ProjectGetDTO add( @RequestBody  ProjectAddDTO DTO) {
+        // mapping DTO to service request model
+        ProjectServiceRequestModel serviceModel = new ProjectServiceRequestModel(DTO.getTitle(), DTO.getPortfolioId());
```
un seul model pour ton service

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/login/login.component.ts:
```diff
@@ -30,10 +30,10 @@ export class LoginComponent implements OnInit {
       },    
       error: (err:Error) => {
-                              this.loginError="Mdp ou email invalides";
+                this.loginError="Mdp ou email invalides";
```
Mot de passe

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/project.component.ts:
```diff
@@ -80,11 +102,18 @@ export class ProjectComponent {
+   * Delete the picture on db
+   */
+  onDeletePicture = () => {
+    console.log("hello");
```
oublie pas de le changer met un TODO dessus

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/services/education.service.ts:
```diff
@@ -18,7 +18,7 @@ export class EducationService {
   ) { }
 
-  saveEducation = ( table: string, newEducation: Education ): Observable<any> => {     
+  saveEducation = ( table: string, newEducation: Education ): Observable<any> => {  
```
le jour ou tu te trompe sur la table string tu vas passer 3 heures à chercher le probleme

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/services/education.service.ts:
```diff
@@ -32,10 +32,9 @@ export class EducationService {
-    return this.http.post(`${this.ENV_DEV}/${table}/${newEducation.id}`, newEducation );
-
+    //  update *******
+    return this.http.put(`${this.ENV_DEV}/${table}/${newEducation.id}`, newEducation );
```
met la en dur ici la table

hugomontes24 committed:
Feat(delete document): Deleting document on project. Back 'n Front.
branch bac  #39

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/login/login.component.ts:
```diff
@@ -33,7 +33,7 @@ export class LoginComponent implements OnInit {
       },    
       error: (err:Error) => {
-                this.loginError="Mdp ou email invalides";
+                this.loginError="Mdp invalide";
```
je voulais te dire ecris Mot de passe pas Mdp

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/portfolio-list-item/portfolio-list-item.component.ts:
```diff
@@ -39,7 +39,9 @@ export class PortfolioListItemComponent implements OnInit {
       .subscribe({
-        next:(response:Portfolio) => { this.portfolio = response }, 
+        next:(response:Portfolio) => { this.portfolio = response;
+                                      console.log(response);
```
pas de console log

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/document-project-model.ts:
```diff
@@ -0,0 +1,11 @@
+export class DocumentProjectModel {
+
+    constructor(
```
pas de constructeur pour les objets en Angular on utilise des interfaces

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/document-project-model.ts:
```diff
@@ -0,0 +1,11 @@
+        public mime: string,
+        public filename: string,
+        public projectId: number        
+     ){  }
```
en plus il est vide

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/project-model.ts:
```diff
@@ -1,3 +1,5 @@
+import { DocumentProjectModel } from "./document-project-model";
+
 export class ProjectModel {
```
meme chose

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/project.component.html:
```diff
@@ -52,15 +52,18 @@ <h6 class="d-flex justify-content-between title">
-        <div class="d-flex justify-content-end align-items-start" *ngIf="newProject.fileName" >
-          <div class="preview">
-            <img [src]="projectService.ENV_PICT + newProject.fileName" alt="">  
+        <div *ngFor="let doc of newProject.documents; let j = index">    
```
tu utilise pas le j, si ?

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/project.component.ts:
```diff
@@ -47,13 +49,14 @@ export class ProjectComponent {
           this.projects = 
-            this.projectService.refreshSkills(this.projects,  // tableau d'affichage
+            this.projectService.addProjectToArray(this.projects,  // tableau d'affichage
                                               new ProjectModel( 
```
{ 
id: data.id ...

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/component/project/project.component.ts:
```diff
@@ -127,5 +126,23 @@ export class ProjectComponent {
+          }
+    });
+    // TODO
+    console.log("hello", documentId);
```
pas de console.log(

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/services/project.service.ts:
```diff
@@ -21,12 +24,15 @@ export class ProjectService {
-  saveProject = ( table: string, newProject: Project, selectedFile: File | null  ): Observable<any> => { 
-    const defaultDate = new Date("1970-01-02"); 
+  saveProject = ( table: string, newProject: Project, selectedFile: File | null ): Observable<any> => { 
+    console.log(newProject);
```
pas de console log

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/services/project.service.ts:
```diff
@@ -21,12 +24,15 @@ export class ProjectService {
-    if( newProject.date < defaultDate){        
+    if(newProject.date == null ){
+      formData.append('date', new Date("1970-01-01").toLocaleDateString()); // date generated by the app (1970-01-01)
+    }else if ( newProject.date < defaultDate){        
```
je sais pas si c'est une bonne idéee

Mekouar-Ayoub commented on ngEasyPortfolio/src/app/portfolio/services/project.service.ts:
```diff
@@ -63,13 +69,17 @@ export class ProjectService {
+  
+  // DOCUMENTS  **************************
+  deleteDocument = (table: string , projectId:number, documentId: number): Observable<any> | any => {
+    return this.http.delete(`${this.ENV_DEV}/${table}/${projectId}/documents/${documentId}` );  
```
evite le ${table} ecris le en dur
